src/core/layout/qgslayoutitemlegend.h

```diff
@@ -44,11 +46,48 @@ class CORE_EXPORT QgsLegendModel : public QgsLayerTreeModel
 
   public:
     //! Construct the model based on the given layer tree
-    QgsLegendModel( QgsLayerTree *rootNode, QObject *parent SIP_TRANSFERTHIS = nullptr );
+    QgsLegendModel( QgsLayerTree *rootNode, QObject *parent SIP_TRANSFERTHIS = nullptr, QgsLayoutItemLegend *layout = nullptr );
+
+    //! Alternative constructor.
+    QgsLegendModel( QgsLayerTree *rootNode,  QgsLayoutItemLegend *layout );
 
     QVariant data( const QModelIndex &index, int role ) const override;
 
     Qt::ItemFlags flags( const QModelIndex &index ) const override;
+
+  signals:
+
+    /**
+     * Emitted to refresh the legend once counting is done.
```

Comment:
```suggestion
     * Emitted to refresh the legend.
```

This could be used for other things than counting in the future, correct?

Reply:
Eventually ,yes.